src/server/saved_objects/saved_objects_mixin.js

```diff
@@ -26,11 +26,30 @@ export function savedObjectsMixin(kbnServer, server) {
   server.route(createGetRoute(prereqs));
   server.route(createUpdateRoute(prereqs));
 
+  async function onBeforeWrite() {
+    const adminCluster = server.plugins.elasticsearch.getCluster('admin');
+
+    try {
+      await adminCluster.callWithInternalUser('cluster.health', {
+        timeout: server.config().get('savedObjects.indexCheckTimeout'),
```

Comment:
Rather than polling for the health of the index, why not just do a simple 404 check on the index itself?  The write request itself doesn't poll or anything, so it doesn't seem especially important to poll for availability before it.

Reply:
The cluster.health call confirms not only that the index exists, but that is also usable. When the index is first created it goes through a phase of being red before it turns yellow and is actually ready for use. This is why we have the same `cluster.health` check in the healthcheck, because indexing or searching for a config doc immediately after the index is created produces its own set of errors.

I also don't think it's especially important to wait for the index, but the idea is that if the healthCheck is left at 2.5 seconds then writes to the savedObjects should just always succeed.

That said, once we have an index template the check will just ensure that the index template exists and be instant.